gapis/api/gles/compat.go

```diff
@@ -601,12 +601,20 @@ func compat(ctx context.Context, device *device.Instance) (transform.Transformer
 				return
 			}
 		case *GlTexStorage2DMultisample:
-			{
+			if version.IsES || version.AtLeastGL(4, 3) {
```

Comment:
I would drop the 4.3 part of the condition.  But that is probably just me.

Reply:
I'd like to steer the compat code towards doing the right checks. My longer term plan of action is to pull this logic out of this mega function and as methods of the command. We can then use these methods check to see if the particular command is supported by the given device - and use this to pick the most suitable device and/or show compatibility warnings in the future.

Reply:
I agree with regard to checks - but in different way - ideally add some support for desktop GL into the api files as well, and then mutate the output through that to verify it should work.  For a set of devices this would tell us which would fail (and how), and which would pass, all without trying to run it on the devices.  As a poor man's version of this you could just record the minimal requirements as you execute the compat.  So even without any replay device you would get transformed stream and the needed min spec.

With regards transforms - I prefer unconditional ones.  That is, simplify complicated GL commands to simpler canonical ones.  This is great example of that.  There is no reason not to always simplify to more supported glTexImage2DMultisample version.  Similarly all other *Storage methods can be simplified to more supported GL commands (although that is a bit more code, and we did not hit that compat need yet).

Anyway, it just my 1 cent.  Feel free to follow your path.  But if you get to situation where you can not reproduce a bug because the user's GL version and set of extensions follows a different path then yours, don't say I didn't warn you :-P  (e.g. if the AtLeast(4,3) part of this block gets accidentally broken in the future as part of refactoring, you will never find out on your machine)

Reply:
Acknowledged.